src/main/java/org/logstash/dissect/Converters.java

```diff
@@ -2,21 +2,24 @@
 
 import org.logstash.Event;
 
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
 public enum Converters implements Converter {
     INT {
         @Override
         public void convert(final Event e, final String src) {
             final Object value = e.getField(src);
-            final long v = new Double(String.valueOf(value)).longValue();
-            e.setField(src, v);
+            BigDecimal bd = new BigDecimal(String.valueOf(value)).setScale(12, RoundingMode.FLOOR);
+            e.setField(src, bd.longValue());
```

Comment:
@guyboertje is `long` actually the type we expect here?
This will have the weird result that numbers larger than `LONG_MAX` will simply be "cut off" and you'll get the lowest 64bit. Def. different from what you'd expect to happen in Ruby isn't it?
Maybe just set the `BigInteger` value here?